accord-core/src/test/java/accord/impl/list/ListRequest.java

```diff
@@ -108,27 +110,39 @@ public void accept(Result success, Throwable fail)
             }
             else if (fail instanceof CoordinateFailed)
             {
+                node.reply(client, replyContext, ListHeartbeat.INSTANCE);
                 ((Cluster)node.scheduler()).onDone(() -> {
                     RoutingKey homeKey = ((CoordinateFailed) fail).homeKey();
                     TxnId txnId = ((CoordinateFailed) fail).txnId();
-                    CheckOnResult.checkOnResult(node, txnId, homeKey, (s, f) -> {
-                        if (f != null)
-                            return;
-                        switch (s)
-                        {
-                            case Invalidated:
-                                node.reply(client, replyContext, new ListResult(client, ((Packet)replyContext).requestId, txnId, null, null, null, null));
-                                break;
-                            case Lost:
-                                node.reply(client, replyContext, new ListResult(client, ((Packet)replyContext).requestId, txnId, null, null, new int[0][], null));
-                                break;
-                            case Neither:
-                                // currently caught elsewhere in response tracking, but might help to throw an exception here
-                        }
-                    });
+                    checkOnResult(homeKey, txnId);
                 });
             }
         }
+
+        private void checkOnResult(RoutingKey homeKey, TxnId txnId) {
+            CheckOnResult.checkOnResult(node, txnId, homeKey, (s, f) -> {
+                if (f != null)
+                {
+                    if (f instanceof Timeout || f instanceof SimulatedFault) checkOnResult(homeKey, txnId);
+                    else node.agent().onUncaughtException(f);
+                    return;
+                }
+                switch (s)
+                {
+                    case Invalidated:
+                        node.reply(client, replyContext, ListResult.invalidated(client, ((Packet)replyContext).requestId, txnId));
+                        break;
+                    case Lost:
+                        node.reply(client, replyContext, ListResult.lost(client, ((Packet)replyContext).requestId, txnId));
+                        break;
+                    case Neither:
+                        node.reply(client, replyContext, ListResult.timeout(client, ((Packet)replyContext).requestId, txnId));
```

Comment:
might have missed something, but when I hit this in burn the comment "// currently caught elsewhere in response tracking, but might help to throw an exception here" didn't seem to reflect what was going on.

What I saw was that the txn was `Applied`, but the client request `Timeout`, so when we tried to figure out the state it is `Neither` which would not cause us to bump the `clock`, which lead to burn failing due to clock mismatch